components/scream/src/share/util/scream_vertical_interpolation.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef SCREAM_VERTICAL_INTERPOLATION_HPP
+#define SCREAM_VERTICAL_INTERPOLATION_HPP
+
+#include "share/scream_types.hpp"
+
+#include "ekat/util/ekat_lin_interp.hpp"
+#include "ekat/ekat_pack_utils.hpp"
+#include "ekat/kokkos/ekat_subview_utils.hpp"
+
+namespace scream {
+
+//
+// ------- Types --------
+//
+
+template <typename S>
+using SmallPack = ekat::Pack<S,SCREAM_SMALL_PACK_SIZE>;
+using Spack = SmallPack<Real>;
+
+template <typename S>
+using SmallPack = ekat::Pack<S,SCREAM_SMALL_PACK_SIZE>;
+using Sbool = SmallPack<bool>;
+
+using Smask = ekat::Mask<Spack::n>;
+
+using KT = KokkosTypes<DefaultDevice>;
+using ExeSpace = typename KT::ExeSpace;
+using ESU = ekat::ExeSpaceUtils<ExeSpace>;
+using LIV = ekat::LinInterp<Real,Spack::n>;
+
+ 
+template <typename S>
+using view_1d = typename KT::template view_1d<S>;
+template <typename S>
+using view_2d = typename KT::template view_2d<S>;
+
+const Real masked_val = -9.99e17;
+  
+void perform_vertical_interpolation (const view_2d<const Spack>&,
```

Comment:
I think these functions could be templated on scalar type and pack size, so that they could potentially be used even if some parametrization used a different pack size.

Reply:
Have modified the functions to be templated on scalar type and pack size. To do source moved function implementation into a file scream_vertical_interpolation_impl.hpp and deleted scream_vertical_interpolation.cpp